Add "must change password" mechanism + Normalize response for auth operations

nb-ohad opened the pull request:
### Purpose
- Allow setting a future date where the account will have to change it's password.
### Checklist
- Code:
  - [ ] User Experience: **Taken a moment to think the effects on our user**
  - [ ] Failure handling and Comments on non trivial sections: 
  - [ ] Cross Platform: **Supporting Win 7/8/10 Server 12, Linux, Mac**
- Testing:
  - [ ] Unit/System Tests: **Added tests ... / Already covered by existing tests ...**
  - [x] Tested with: `npm test`
- Supportability:
  - [ ] Diagnostics info, Phone Home info, ActivityLog events, External Syslog: 
- Upgrade:
  - [ ] Mongo Schema Upgrade:
  - [ ] Agents changes, Server Platform changes and New packages added to package.json:

nb-ohad committed:
Add "must change password" mechanism to BE

nb-ohad committed:
Cange default create system password policy

Until FE implementation of change password window

guymguym reviewed (changes requested):
fix `idate` usage (see comments)

nb-ohad committed:
Merge branch 'master' into ohad-frontend-dev

nb-ohad committed:
Normalize account_info between read_auth and create_auth

nimrod-becker reviewed (approved):
Approve for the "Normalize account_info between read_auth and create_auth" part

nb-ohad committed:
Merge branch 'master' into ohad-frontend-dev

nb-ohad committed:
Add date format to son schema, allow validating date in database schema

Covert all known uses of Date object in the database from idate to date.

Future action is needed to convert all other dates (that actually save
an integer in the database) into date format

nb-ohad committed:
Fix some wrong changes from last commit
